src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -335,6 +335,11 @@ internal static void EmitKeyedServiceParameterPreparation(this EndpointParameter
     {
         codeWriter.WriteLine(endpointParameter.EmitParameterDiagnosticComment());
 
+        codeWriter.WriteLine("if (httpContext.RequestServices.GetService<IServiceProviderIsService>() is not IServiceProviderIsKeyedService)");
+        codeWriter.StartBlock();
+        codeWriter.WriteLine(@"throw new InvalidOperationException(""This service provider doesn't support keyed services."");");
```

Comment:
What about changing the error message to something like:

> Unable to resolve {nameof(FromKeyedServicesAttribute)} because the service provider doesn't support keyed services.

That gives the developer a clue about the cause and effect.